master/internal/workspace/authz_basic_impl.go

```diff
@@ -158,6 +158,13 @@ func (a *WorkspaceAuthZBasic) CanCreateWorkspaceWithCheckpointStorageConfig(
 	return nil
 }
 
+// CanSetWorkspacesDefaultPools returns a nil error.
+func (a *WorkspaceAuthZBasic) CanSetWorkspacesDefaultPools(
+	ctx context.Context, curUser model.User, workspace *workspacev1.Workspace,
```

Comment:
Since we'll almost definitely never need anything except the user's id, I think this function should accept just the user id. I'm imagining a case down the road of somebody having a user id handy but needing to construct an entire `User` stuct, only to have the function throw away everything but the id.